### DISCUSSION TITLE
ConsolePi on BananaPi M2 Zero

### DISCUSSION THREAD

**Discussion by phredi-renner** [Pack3tL0ss/ConsolePi#157] [Show and tell]
Since all things RaspberryPi are hard to come by and super pricey these days, if they are even available, I gave ConsolePi a try on a BananaPi M2 Zero running Ubuntu 20.04 LTS (focal) and was successful.

**Comment by n3gwg** [comment]
Do you think this project would work on other Raspberry Pi clones?

**Reply by phredi-renner** [reply]
It should, however it does require the rpi.gpio module and clones like the Orange Pi use opi.gpio and that did not work when I tried it. The BananaPi seems to be a closer clone to RaspberryPi.

**Reply by n3gwg** [reply]
Oh! I thought consolePi could work with USB based serial ports ostensibly negating the requisite for using gpio based serial connections. I have to admit, I have not yet had time to review the source.

I don't have a Pi available to dedicate for this purpose at this time. My hope was, either to get an inexpensive Pi 3B (or similar clone) or that ConsolePi could be deployed in a Linux virtual machine, and that I can pass through some USB serial ports to it.

Thanks for your guidance I'll back burner playing with this for now.

Stuart

**Reply by Pack3tL0ss** [reply]
I'll review the logic around the RPi.GPIO.  It installs it on all systems, but it's only there for optional power management, (operating relays connected to GPIO that control power outlets).  Which is likely the less frequently used option for that (ConsolePi supports some other network based outlets as well, which is more common).  Currently it installs it regardless of platform, but if it can't run it just disables GPIO based outlet control.
